Sources/Knit/Module/ModuleAssembly.swift

```diff
@@ -40,6 +45,24 @@ public extension ModuleAssembly {
             return self.resolverType == $0.resolverType
         }
     }
+
+    static var _assemblyFlags: [ModuleAssemblyFlags] {
+        var result: [ModuleAssemblyFlags] = []
+        if self is any AutoInitModuleAssembly.Type {
+            result.append(.autoInit)
+        }
+        if self is any AbstractAssembly.Type {
```

Comment:
It seems like for these `is` checks we are just moving the call site? Is there some other way that this causes a performance improvement?

Reply:
This is the fallback, it only gets hit if a code gen version isn't available.